src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -122,7 +123,18 @@ public boolean verifyUser(String username, String uniqueid) {
 
     @Override
     public boolean canUseLocalCache() {
-        boolean brokerSwitch = canSwitchToBroker();
+        boolean brokerSwitch;
+        try
+        {
+            brokerSwitch = canSwitchToBroker();
+        }
+        catch (DeveloperAuthenticationException e){
```

Comment:
I'm picky here. But do you need to change the thrown exception? can we make it final?